src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Tests.Common.cs

```diff
@@ -154,6 +154,22 @@ public static async Task<Regex[]> GetRegexesAsync(RegexEngine engine, params (st
             RegexEngine.NonBacktrackingSourceGenerated => RegexOptionNonBacktracking | RegexOptions.Compiled,
             _ => throw new ArgumentException($"Unknown engine: {engine}"),
         };
+
+        /// <summary>Set the AppContext variable REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE to the given max value. Only used with Nonbacktracking engine.</summary>
+        public static void SetSafeSizeThreshold(int maxSize)
+        {
+#if NET7_0_OR_GREATER
+            AppContext.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", maxSize);
+#endif
+        }
+
+        /// <summary>Remove the AppContext variable REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE value. Only used with Nonbacktracking engine.</summary>
+        public static void RestoreSafeSizeThresholdToDefault()
+        {
+#if NET7_0_OR_GREATER
+            AppContext.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", null);
+#endif
```

Comment:
```suggestion
            AppContext.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", null);
```

Reply:
Will not compile if I remove the `ifdef`.